python/ripgrep.py

```diff
@@ -0,0 +1,280 @@
+        weechat.config_get("irc.color.nick_prefixes")
+    )
+
+
+def get_logfile_from_buffer_pointer(buffer: str):
+    infolist = weechat.infolist_get("logger_buffer", "", "")
+    if infolist:
+        while weechat.infolist_next(infolist):
+            pointer = weechat.infolist_pointer(infolist, "buffer")
+            if pointer == buffer:
+                file = weechat.infolist_string(infolist, "log_filename")
+                weechat.infolist_free(infolist)
+                return file
+        weechat.infolist_free(infolist)
+
+
+def get_or_create_buffer(title: str = None):
+    buffer = weechat.buffer_search("python", SCRIPT_NAME)
+    if not buffer:
+        buffer = weechat.buffer_new(SCRIPT_NAME, "", "", "", "")
+        weechat.buffer_set(buffer, "time_for_each_line", "1")
+        weechat.buffer_set(buffer, "nicklist", "0")
+        weechat.buffer_set(
+            buffer,
+            "title",
+            title or "ripgrep.py | use \\rg to search through buffer logfiles",
+        )
+        weechat.buffer_set(buffer, "localvar_set_no_log", "1")
+    if title:
+        weechat.buffer_set(buffer, "title", title)
+    return buffer
+
+
+def get_color_for_mode(mode):
+    if mode not in "~@!%+":
+        return
+
+    idx = "~@!%+".find(mode)
+    if idx == -1:
+        return
+    key = "qoahv"[idx]
+
+    for mapping in CONFIG["IRC_COLOR_NICK_PREFIXES"].split(";"):
+        if mapping.startswith(key):
+            return weechat.color(mapping.split(":")[1])
+
+
+def buffer_print_match(buffer, match):
+    shift = 0
+    fulltext = match["data"]["lines"]["text"]
+    try:
+        dt, nick, msg = fulltext.split("\t", 2)
+    except ValueError as ex:
+        print(ex, "\t", fulltext)
+        return
+
+    # parse time str to unix timestamp
+    try:
+        timestamp = int(
+            time.mktime(time.strptime(dt, CONFIG["LOGGER_FILE_TIME_FORMAT"]))
+        )
+    except ValueError:
+        # if we couldn't parse dt the time_format was probably changed
+        timestamp = 0
+
+    colorize_nick = False
+
+    # ACTION
+    if nick.strip() == "*":
+        nick = msg.split()[0]
+    # NOTICE
+    elif nick == CONFIG["WEECHAT_LOOK_PREFIX_NETWORK"] and msg.startswith("Notice("):
+        nick = msg.split("(", 1)[1].split(")", 1)[0]
+    # JOIN | PART
+    elif (
+        nick == CONFIG["WEECHAT_LOOK_PREFIX_JOIN"]
+        or nick == CONFIG["WEECHAT_LOOK_PREFIX_JOIN"]
+    ):
+        nick = msg.split()[0]
+    else:
+        # TODO: currently we only colorize in privmsgs
+        colorize_nick = True
+
+    # separate channel mode from nickname
+    try:
+        if nick[0] in "~@!%+":
+            nick = nick.lstrip("@!%+")
+    except IndexError:
+        pass
+
+    color_highlight = weechat.color("red")
+    color_default = weechat.color("chat")
+    color_nick = weechat.info_get("irc_nick_color", nick) or 0
+    try:
+        color_nick_number = int(color_nick.replace("\x19F", "", 1)) or 0
```

Comment:
```suggestion
        color_nick_number = int(color_nick.replace("\x19F@", "", 1)) or 0
```

Without this change, I get an error on WeeChat 2.6:
```
13:25:30 python: stdout/stderr (ripgrep): Traceback (most recent call last):
13:25:30 python: stdout/stderr (ripgrep):   File "/home/nyuszika7h/.weechat/python/ripgrep.py", line 241, in rg_cb
13:25:30 python: stdout/stderr (ripgrep):     buffer_print_match(buffer, result)
13:25:30 python: stdout/stderr (ripgrep):   File "/home/nyuszika7h/.weechat/python/ripgrep.py", line 134, in buffer_print_match
13:25:30 python: stdout/stderr (ripgrep):     color_nick_number = int(color_nick.replace("\x19F", "", 1)) or 0
13:25:30 python: stdout/stderr (ripgrep): ValueError: invalid literal for int() with base 10: '@00130'
```

Reply:
Try `/rg grep` in #weechat. Fails for me like this with your change:

```py3tb
13:57 python: stdout/stderr (ripgrep): Traceback (most recent call last):
13:57 python: stdout/stderr (ripgrep):   File "/var/lib/weechat/python/ripgrep.py", line 241, in rg_cb
13:57 python: stdout/stderr (ripgrep):     buffer_print_match(buffer, result)
13:57 python: stdout/stderr (ripgrep):   File "/var/lib/weechat/python/ripgrep.py", line 134, in buffer_print_match
13:57 python: stdout/stderr (ripgrep):     color_nick_number = int(color_nick.replace("\x19F@", "", 1)) or 0
13:57 python: stdout/stderr (ripgrep): ValueError: invalid literal for int() with base 10: '\x19F11'
13:57 ⚠  python: error in function "rg_cb"
```

I guess I have to read up on the documentation of color codes in weechat. The currently implementation is naïve in that regard.

Reply:
```suggestion
        color_nick_number = int(color_nick.replace("\x19F", "", 1).lstrip("@")) or 0
```

Seems like some nick colors are like `@00062` while others are just like `10`. This should do the trick.

Reply:
Applied, thank you!